pkg/controller/inferenceservice/reconcilers/istio/virtualservice_reconciler.go

```diff
@@ -60,6 +62,10 @@ func (r *VirtualServiceReconciler) Reconcile(isvc *v1alpha2.InferenceService) er
 		return fmt.Errorf("failed to reconcile virtual service: desired and status are nil")
 	}
 
+	if err := r.reconcileExternalService(isvc); err != nil {
```

Comment:
Can you explain this addition?

Reply:
When you call inference service virtual service directly from internal mesh there is no kubernetes service currently, it works outside because it matches the external host and route properly via virtual service to the actual predictor kubernetes service. So we need to create the service here for the virtual service host e.g `mnist.default.svc.cluster.local` and it has to be external service type because it redirects to istio local cluster gateway in `istio-system` namespace.